server/src/main/java/io/unitycatalog/server/service/VolumeService.java

```diff
@@ -3,52 +3,170 @@
 import com.linecorp.armeria.common.HttpResponse;
 import com.linecorp.armeria.common.HttpStatus;
 import com.linecorp.armeria.server.annotation.*;
+import io.unitycatalog.server.auth.UnityCatalogAuthorizer;
+import io.unitycatalog.server.auth.annotation.AuthorizeExpression;
+import io.unitycatalog.server.auth.annotation.AuthorizeKey;
+import io.unitycatalog.server.auth.annotation.AuthorizeKeys;
+import io.unitycatalog.server.auth.decorator.UnityAccessEvaluator;
+import io.unitycatalog.server.auth.decorator.UnityAccessUtil;
 import io.unitycatalog.server.exception.GlobalExceptionHandler;
+import io.unitycatalog.server.model.CatalogInfo;
 import io.unitycatalog.server.model.CreateVolumeRequestContent;
+import io.unitycatalog.server.model.ListVolumesResponseContent;
+import io.unitycatalog.server.model.Privilege;
+import io.unitycatalog.server.model.SchemaInfo;
 import io.unitycatalog.server.model.UpdateVolumeRequestContent;
+import io.unitycatalog.server.model.VolumeInfo;
+import io.unitycatalog.server.persist.CatalogRepository;
+import io.unitycatalog.server.persist.MetastoreRepository;
+import io.unitycatalog.server.persist.SchemaRepository;
 import io.unitycatalog.server.persist.VolumeRepository;
+import lombok.SneakyThrows;
+
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
+import java.util.UUID;
+
+import static io.unitycatalog.server.model.ResourceType.CATALOG;
+import static io.unitycatalog.server.model.ResourceType.METASTORE;
+import static io.unitycatalog.server.model.ResourceType.SCHEMA;
+import static io.unitycatalog.server.model.ResourceType.TABLE;
+import static io.unitycatalog.server.model.ResourceType.VOLUME;
 
 @ExceptionHandler(GlobalExceptionHandler.class)
 public class VolumeService {
   private static final VolumeRepository VOLUME_REPOSITORY = VolumeRepository.getInstance();
+  private static final SchemaRepository SCHEMA_REPOSITORY = SchemaRepository.getInstance();
+  private static final CatalogRepository CATALOG_REPOSITORY = CatalogRepository.getInstance();
 
-  public VolumeService() {}
+  private final UnityCatalogAuthorizer authorizer;
+  private final UnityAccessEvaluator evaluator;
+
+  @SneakyThrows
+  public VolumeService(UnityCatalogAuthorizer authorizer) {
+    this.authorizer = authorizer;
+    evaluator = new UnityAccessEvaluator(authorizer);
+  }
 
   @Post("")
-  public HttpResponse createVolume(CreateVolumeRequestContent createVolumeRequest) {
+  // TODO: for now, we are not supporting CREATE VOLUME or CREATE EXTERNAL VOLUME privileges
+  @AuthorizeExpression("""
+          #authorize(#principal, #metastore, METASTORE_ADMIN) ||
+          #authorizeAny(#principal, #catalog, OWNER, USE_CATALOG) && #authorizeAny(#principal, #schema, OWNER, USE_SCHEMA)
+          """)
```

Comment:
I don't think I fully understand how these annotations and expressions are working, will setup some time to discuss